packages/taco-auth/src/types.ts

```diff
@@ -1,9 +1,33 @@
-import { FormattedEIP4361 } from './eip4361';
-import { FormattedEIP712 } from './eip712';
+import {EIP4361AuthProvider, EIP4361TypedData} from './providers/eip4361';
+import {EIP712AuthProvider, EIP712TypedData} from './providers/eip712';
+
+export interface AuthProvider {
+  getOrCreateAuthSignature(): Promise<AuthSignature>;
+}
+
+export type AuthProviders = {
+  [EIP712_AUTH_METHOD]?: EIP712AuthProvider;
+  [EIP4361_AUTH_METHOD]?: EIP4361AuthProvider;
+  // Fallback to satisfy type checking
+  [key: string]: AuthProvider | undefined;
+};
 
 export interface AuthSignature {
   signature: string;
   address: string;
   scheme: 'EIP712' | 'EIP4361';
```

Comment:
Is it helpful for these values to use the `*_AUTH_METHOD` constants below?

Reply:
I'm afraid it's not so easy with TypeScripts, types and constants do not mix.